libcxx/include/__new/operators.h

```diff
@@ -0,0 +1,100 @@
+//===----------------------------------------------------------------------===//
```

Comment:
Myybe name this file `operator_new.h`?

Reply:
This also defines `operator delete`, though. Does that change your preference?

Reply:
I'm not a huge fan of `operators.h`, since we have quite a few very similar headers which have nothing actually in common (`{__exception,__filesystem,__functional}/operations.h`). I realize that I'm at least partially at fault, but if we can find a better name it'd be good to avoid IMO.

Thinking a bit more about it, maybe we could have `global_new_delete.h` and `placement_new_delete.h`. They don't really have much to do with each other and are used in quite different places usually.

Reply:
How about simply `operator_new_delete.h`?

We generally use `std::__construct_at` instead of direct placement-new, so IMO there's not that much value in separating both into separate headers.

Reply:
I'd be fine with that as well, but note that `__construct_at` uses placement new, so moving that into its own header would make `construct_at.h` quite a bit smaller.